net/raft/raft_test.go

```diff
@@ -141,6 +141,63 @@ func TestClusterSetup(t *testing.T) {
 	}
 }
 
+func TestNodeRestart(t *testing.T) {
+	ctx := context.Background()
+
+	// Create new test cluster
+	nodeA, nodeB, nodeC := newTestCluster(ctx, t)
+	defer nodeA.cleanup()
+	defer nodeB.cleanup()
+	defer nodeC.cleanup()
+
+	httpClient := nodeA.service.client
+
+	_, err := nodeA.service.Exec(ctx, set("/foo", "bar"))
+	must(t, err)
+
+	// Stop the node
+	nodeA.service.Stop()
+
+	// Restart the node
+	nodeA.service, err = Start(nodeA.addr, nodeA.dir, httpClient, nodeA.state)
```

Comment:
I think we might need to use `newTestState()` here instead of `nodeA.state`. Currently, `Start` expects the state to be the empty state. Then it's the net/raft package's responsibility to call RestoreSnapshot or Apply to bring it up-to-date if there's existing state in persisted in a snapshot or the WAL.